starter-token/checkpoint-2/src/main.nr

```diff
@@ -0,0 +1,127 @@
+use aztec::macros::aztec;
+
+#[aztec]
+pub contract StarterToken {
+    use aztec::{
+        state_vars::{private_set::PrivateSet, public_mutable::PublicMutable, map::Map},
+        messages::logs::note::encode_and_encrypt_note_unconstrained,
+        note::note_viewer_options::NoteViewerOptions,
+        macros::{
+            functions::{initializer, private, public, utility, internal},
+            storage::storage,
+        },
+        protocol_types::address::AztecAddress,
+    };
+
+    #[storage]
+    struct Storage<Context> {
+        balances: Map<AztecAddress, PublicMutable<u128, Context>, Context>,
+        owner: PublicMutable<AztecAddress, Context>,
+        // ===============
+        private_balances: Map<AztecAddress, PrivateSet<UintNote, Context>, Context>,
+    }
+
+    #[initializer]
+    #[public]
+    fn setup() {
+        // The deployer (msg_sender) becomes the owner
+        storage.owner.write(context.msg_sender());
+    }
+
+    #[public]
+    fn mint(to: AztecAddress, amount: u128) {
+        _assert_is_owner(context.msg_sender());
+
+        let recipient_balance = storage.balances.at(to).read();
+
+        storage.balances.at(context.msg_sender()).write(recipient_balance + amount);
+    }
+
+    #[public]
+    fn transfer(to: AztecAddress, amount: u128) {
+        let sender = context.msg_sender();
+
+        let sender_balance = storage.balances.at(sender).read();
+
+        assert(sender_balance >= amount, "Cannot transfer more than the balance of the user");
+
+        storage.balances.at(sender).write(sender_balance - amount);
+
+        let recipient_balance = storage.balances.at(to).read();
+
+        storage.balances.at(to).write(recipient_balance + amount);
+    }
+
+    #[contract_library_function]
+    fn _assert_is_owner(storage: Storage, maybe_owner: AztecAddress) {
```

Comment:
do we need this function defined twice in the contract? once as a library function and once as a public internal function? 

If so, it would be good to explain why.